src/CommunityToolkit.Maui/Extensions/OnScreenSizeExtension.shared.cs

```diff
@@ -0,0 +1,152 @@
+		set => categoryPropertyValues[ScreenCategories.Small] = value;
+	}
+	
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.Medium"/>
+	/// </summary>
+	public object Medium
+	{
+		get => categoryPropertyValues[ScreenCategories.Medium]!;
+		set => categoryPropertyValues[ScreenCategories.Medium] = value;
+	}
+
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.Large"/>
+	/// </summary>
+	public object Large
+	{
+		get => categoryPropertyValues[ScreenCategories.Large]!;
+		set => categoryPropertyValues[ScreenCategories.Large] = value;
+	}
+
+	/// <summary>
+	/// Defines a value used when the screen size is categorized as <see cref="ScreenCategories.ExtraLarge"/>
+	/// </summary>
+	public object ExtraLarge
+	{
+		get => categoryPropertyValues[ScreenCategories.ExtraLarge]!;
+		set => categoryPropertyValues[ScreenCategories.ExtraLarge] = value;
+	}
+
+
+	/// <summary>
+	/// Xaml internal usage
+	/// </summary>
+	public object ProvideValue(IServiceProvider serviceProvider)
+	{
+		var valueProvider = serviceProvider?.GetService<IProvideValueTarget>() ?? throw new ArgumentException($"Service provided for OnScreenSize is null");
+
+		BindableProperty bp;
+		PropertyInfo pi = null!;
+		
+		if (valueProvider.TargetObject is Setter setter)
+		{
+			bp = setter.Property;
+		}
+		else
+		{
+			bp = (valueProvider.TargetProperty as BindableProperty)!;
+			pi = (valueProvider.TargetProperty as PropertyInfo)!;
+		}
+
+		if (Manager.Current.IsLogEnabled)
+		{
+			LogHelpers.WriteLine($"Providing Value using propertyType:\"{(bp?.ReturnType ?? pi?.PropertyType ?? null)}\" and BindableProperty:{(bp ?? null)}", LogLevels.Verbose);
+		}
+		
+		var propertyType = bp?.ReturnType ?? pi?.PropertyType ?? throw new InvalidOperationException("Não foi posivel determinar a propriedade para fornecer o valor.");
+
+		var value = GetScreenCategoryPropertyValue(serviceProvider);
+
+
+		return value!.ConvertTo(propertyType, bp!);
+	}
+
+	/// <summary>
+	/// Gets a value from one of the properties that best suites a <see cref="ScreenCategories"/> a device fits in.
+	/// </summary>
+	/// <param name="serviceProvider"></param>
+	/// <returns></returns>
+	/// <exception cref="XamlParseException"></exception>
+#pragma warning disable IDE0040
+	private object GetScreenCategoryPropertyValue(IServiceProvider serviceProvider)
+#pragma warning restore IDE0040
+	{
+		var screenCategory = ScreenCategoryHelper.GetCategory();
+		if (screenCategory != ScreenCategories.NotSet)
+		{
+			if (categoryPropertyValues[screenCategory] != defaultNull)
+			{
+				return categoryPropertyValues[screenCategory]!;
+			}
+		}
+
+		if (Default == defaultNull)
+		{
+			throw new XamlParseException(string.Format("{0} requires property {0}.{1} defined to use as fallback as property {0}.{2} was not set.",nameof(OnScreenSizeExtension),nameof(Default), screenCategory.ToString()));
```

Comment:
You can, but as this class is a markup extension (it inherits from IMarkupExtension) which in turn  is executed only by "maui xaml engine" it does not make sense to be different.

Reply:
Can we use string interpolation instead of `string.Format`? The reason is that in recent C# versions the string interpolation get optimized.

Reply:
Sure we can, actually if you look old versions of my original code it was string interpolation, I changed because I wanted to put all strings on a resource file, but I endup forgetting that :)

Reply:
No need to care about resource files, we don't localize our project